core/src/main/java/io/debezium/connector/cassandra/CassandraConnectorConfig.java

```diff
@@ -347,11 +347,22 @@ public static EventOrderGuaranteeMode parse(String value) {
             .withType(Type.STRING)
             .withDescription("The absolute path of the YAML config file used by a Cassandra node.");
 
+    public static final Field CASSANDRA_CLUSTER_NAME = Field.create("cassandra.cluster.name")
+            .withType(Type.STRING)
+            .withDescription("Name of Cassandra cluster.")
+            .withDefault("Test Cluster");
```

Comment:
Why it was not necessary to override it before?

Reply:
It is used in production. In `CassandraConnectorTaskTemplate` in `initProcessorGroup`. Before that, there was `deserializerProvider.getClusterName()` at `taskContext.getClusterName()` place.

````
processorGroup.addProcessor(new SnapshotProcessor(taskContext, taskContext.getClusterName()));
````

Now if you think, about that, what the hell does `CassandraTypeProvider` have in common with name of the cluster? It was implemented like this:

````
    @Override
    public String getClusterName() {
        return DatabaseDescriptor.getClusterName();
    }
````

So, the only reason we put this into `CassandraTypeProvider` before was that we were trying to "hide" this _somewhere_ so `DatabaseDescriptor` is on the class path and it is module specific. We could not put this into "core" because `DatabaseDescriptor` is not there (we are not depending on any specific Cassandra version there). 

Cluster name should be fetched from `CassandraConnectorContext`. We have one implementation of the context - `DefaultCassandraConnectorContext` which is in `core`, in `CassandraConnectorTaskTemplate`. So we can not put there `DatabaseDescriptor.getClusterName` because that class is not in `core`. Hence I made a configuration property out of that.

The same logic holds for the other property.

Reply:
@smiklosovic We've also `createInternal` to create field definition. So it might be fine to use it and just add proper description.

Reply:
@jpechane I created https://issues.redhat.com/browse/DBZ-8373

I have removed it and put it back to CassandraTypeProvider even it does not make sense.

Reply:
@jpechane ah, I noticed you comment just now ... I am not completely sure if `createInternal` is the right thing to do. It is not _internal_. I think that property (cluster name) which is used in `SnapshotProcessor` uses name of cluster internally etc so I think that property is really something a user should have a control over.

Reply:
So, lte me rephrase it. If it is something that user can and should use then let's keep it as is. I don't like the default value but it is apparently matching the one in driver so I am going to accept it.
BTW, there slight misunderstanding. I did not ask originally for the removal, just explanation.

Reply:
@jpechane I am fine to rework it in DBZ-8373 just to be more explicit about that. If it was delivered with this patch then it would be all burried in this big patch. Let's just deliver this as it is now and then we can refactor in the other one. That is totally fine with me.